language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -194,6 +180,64 @@ object CodeGenRunner extends StrictLogging {
     )
   }
 
+  private def signatureMapAndPackagePrefixes(
+      darFiles: Iterable[(Path, Option[String])]
+  ): (Map[PackageId, PackageSignature], Map[PackageId, String]) = {
+    val signaturesToPrefixes: Seq[(Option[String], Seq[PackageSignature])] = (for {
+      (path, maybePrefix) <- darFiles
+      signatures = decodeDarAt(path)
+    } yield maybePrefix -> signatures).toSeq
+
+    val packagePrefixes = uniquePackageIdToPrefix(signaturesToPrefixes) ++
+      mainPackageIdToPrefix(signaturesToPrefixes)
+
+    val signatureMap = (for {
+      (_, signatures) <- signaturesToPrefixes
+      signature <- signatures
+    } yield signature.packageId -> signature).toMap
+    signatureMap -> packagePrefixes
+  }
+
+  private def uniquePackageIdToPrefix(
+      signaturesAndPrefixes: Seq[(Option[String], Seq[PackageSignature])]
+  ): Map[PackageId, String] = {
+    val packageIdsAndPackagePrefixes = (for {
+      (packagePrefix, signatures) <- signaturesAndPrefixes
+      if packagePrefix.isDefined
+      signature <- signatures
+    } yield signature.packageId -> packagePrefix.get)
```

Comment:
A pattern on the left of `<-` is always a filter.

```suggestion
      (Some(packagePrefix), signatures) <- signaturesAndPrefixes
      signature <- signatures
    } yield signature.packageId -> packagePrefix)
```